### PR TITLE
Makeover for CI/CD workflows, pre-commit & MkDocs

### LINKED ISSUE #481: Make workflows dispatchable
The CI/CD workflows should be updated. They should be dispatchable (manually executable). Probably using the [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd) re-usable workflows, where possible, and just cleaned up in general.

### LINKED ISSUE #482: Update pypi-release github action
See [this warning](https://github.com/emmo-repo/EMMOntoPy/actions/runs/3180517396/jobs/5184185801#step:16:23). The action should not be using `master` any more.

### LINKED ISSUE #486: Drop Python 3.6 support - extend Python >3.7 support
As seen [here](https://github.com/emmo-repo/EMMOntoPy/actions/runs/3197601428/jobs/5221032835), when running PyTest for EMMOntoPy in a Python 3.6 environment, it is impossible to install the dependencies as they are setup now.

Since Python 3.6 is no longer supported (support ended [9 months ago](https://endoflife.date/python)), I think it's fine to push up EMMOntoPy support to require Python 3.7 at minimum?
If this is not desired, we must put some effort into supporting all - or most - of the EMMOntoPy features in a Python 3.6 environment. At the very least making it installable.

It might also be nice to start testing EMMOntoPy functionality in newer Python versions: 3.8, 3.9, 3.10?
Especially considering Python 3.7 support is stopping [within 8 months](https://endoflife.date/python).

### PR DESCRIPTION
**Pull request by CasperWA**
> See [this comment](https://github.com/emmo-repo/EMMOntoPy/pull/485#issuecomment-1273416213) for an overview of the merging process.

# Description:

Fixes #481 
Fixes #482 
Closes #486 

### CI/CD

This PR implements a complete makeover for the CI/CD workflows, utilizing the [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd) callable GitHub Actions workflows as well as the pre-commit hooks there.

Benefits of this makeover:

- Automatic updates of the underlying workflows.
- Remove `invoke` dependency and `tasks.py` file, since these `invoke` tasks are replaced by the hooks and workflows from SINTEF/ci-cd.
- Simplified workflow file contents, where only a few jobs are now explicit. Those, which are very EMMOntoPy-specific.

Possible drawbacks:

- "Outsourcing" workflows and formerly internal `invoke` tasks for pre-commit hooks can lead to less flexibility if changes are needed and not implemented in the external workflows/hooks.
- Decreased transparency. One now has to inspect the external workflows/hooks to understand _exactly_ what is going on.

Overall, I think this is a huge improvement and should ease CI/CD immensely.

### Documentation

Some updates in the MkDocs configuration were necessary to fix some presentation issues of certain documentation pages.
The table of contents (ToC) has been improved to exclude the `emmodoc` markdown template files under the "Examples" section.

The EMMC-CSA reference on the landing page has been updated to point to emmc.eu instead of the old emmc.info web page. Furthermore, the logo reference has been fixed to point to a local PNG image.

## Type of change:

- [x] Bug fix.
- [ ] New feature.
- [x] Documentation update.

## Checklist:

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

**Comment by CasperWA** [comment]
This is a draft until #486 has been resolved.

File: docs/index.md [hunk 1]
```diff
@@ -144,9 +143,9 @@ It has mainly been developed by [SINTEF](https://www.sintef.no/), specifically:
 
 ### Contributing projects
 
-- [EMMC-CSA](https://emmc.info/about-emmc-csa/);
+- [EMMC-CSA](https://emmc.eu/);
```
Comment (francescalb) [inline docs/index.md:146]: Same as above

File: README.md [hunk 2]
```diff
@@ -144,9 +143,9 @@ It has mainly been developed by [SINTEF](https://www.sintef.no/), specifically:
 
 ### Contributing projects
 
-- [EMMC-CSA](https://emmc.info/about-emmc-csa/);
+- [EMMC-CSA](https://emmc.eu/);
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply README.md:146]: Sure. But... 
image

[the remark this answers is not in the document]
Comment (CasperWA) [reply README.md:146]: I think they've closed emmc.info down in favor of emmc.eu. Then I'd rather link to the new one - perhaps an "about EMMC" to be completely specific?

File: docs/index.md (hunk 1 above)
Comment (CasperWA) [reply docs/index.md:146]: See conversation/comment thread above.

**Review by francescalb — CHANGES_REQUESTED** [review]
I think you suggestion of removing official support of python 3.6 is good. We have not been testing for that version until now. Alternatively, we have to rename the excel-file to .xls. I think that should work, but it is getting quite outdated.

**Comment by CasperWA** [comment]
I have removed Python 3.6 support, while extending official support to include Python 3.8, 3.9, and 3.10 (by adding these versions specifically to the metadata in `setup.py` - and also running the test suite with these versions.

**Comment by CasperWA** [comment]
@jesper-friis I'd like some input /review from you on this PR as well, especially regarding the supported Python version changes this will introduce (see #486 for reference).

**Review by francescalb — APPROVED** [review]
This looks good. I am however, a little hesitant about it being much less transparent now. I suggest we open an issue about documenting a bit more.

**Comment by CasperWA** [comment]
Information on the finalization of this PR:
1. 2 approvals required
2. Branch protection rules update for `master` (and `ci/dependabot-updates`) to "require" the new/updated) workflows
3. Merge
4. Update other existing PRs with the newly merged updates to `master`

**Comment by CasperWA** [comment]
> This looks good. I am however, a little hesitant about it being much less transparent now. I suggest we open an issue about documenting a bit more.

That's completely fair.
I have foreseen this (slightly) and applied myself to try and write some nice documentation for the workflows and pre-commit hooks I'm using here, which are coming from the [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd) GitHub repository.
The documentation can be found [here](https://SINTEF.github.io/ci-cd).

But we should definitely extend the EMMOntoPy documentation for developers to include references to this other documentation.

**Comment by jesper-friis** [comment]
> @jesper-friis I'd like some input /review from you on this PR as well, especially regarding the supported Python version changes this will introduce (see #486 for reference).

I agree with this suggestion.

**Comment by jesper-friis** [comment]
> > This looks good. I am however, a little hesitant about it being much less transparent now. I suggest we open an issue about documenting a bit more.
> 
> That's completely fair. I have foreseen this (slightly) and applied myself to try and write some nice documentation for the workflows and pre-commit hooks I'm using here, which are coming from the [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd) GitHub repository. The documentation can be found [here](https://SINTEF.github.io/ci-cd).
> 
> But we should definitely extend the EMMOntoPy documentation for developers to include references to this other documentation.

If we could improve the documentation of [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd), I think delegating to those tools is only beneficially. See e.g. [cache](https://github.com/marketplace/actions/cache#usage) for a typical example of documentation a github action.

**Comment by CasperWA** [comment]
> > > This looks good. I am however, a little hesitant about it being much less transparent now. I suggest we open an issue about documenting a bit more.
> > 
> > 
> > That's completely fair. I have foreseen this (slightly) and applied myself to try and write some nice documentation for the workflows and pre-commit hooks I'm using here, which are coming from the [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd) GitHub repository. The documentation can be found [here](https://SINTEF.github.io/ci-cd).
> > But we should definitely extend the EMMOntoPy documentation for developers to include references to this other documentation.
> 
> If we could improve the documentation of [SINTEF/ci-cd](https://github.com/SINTEF/ci-cd), I think delegating to those tools is only beneficially and makes the ci/cd much easier to read. See e.g. [cache](https://github.com/marketplace/actions/cache#usage) for a typical example of documentation a github action.

It's important to understand that the SINTEF/ci-cd repository does _not_ provide actions, but rather _callable workflows_. I.e., complete workflows with it's own step-logic and actions usage.
Furthermore, it provides pre-commit hooks to autoupdate API reference documentation and the landing page when using the MkDocs framework.

I think the documentation is quite extensive as is, but everything can always be improved. Please visit the documentation [here](https://SINTEF.github.io/ci-cd) and evaluate for yourself.

**Comment by francescalb** [comment]
> 

I think this looks quite OK already. Only thing is to add the documentation (link to this repo) for ontopy developers as you mention.

**Pull request merged**